suite-common/wallet-core/src/device/deviceReducer.ts

```diff
@@ -586,17 +586,14 @@ export const prepareDeviceReducer = createReducerWithExtraDeps(initialState, (bu
         .addCase(deviceAuthenticityActions.result, (state, { payload }) => {
             setDeviceAuthenticity(state, payload.device, payload.result);
         })
-        .addCase(deviceActions.dismissFirmwareRevisionCheck, (state, { payload }) => {
+        .addCase(deviceActions.dismissFirmwareAuthenticityCheck, (state, { payload }) => {
             if (!state.dismissedSecurityChecks) {
                 state.dismissedSecurityChecks = {};
             }
-            if (!state.dismissedSecurityChecks.firmwareRevision) {
-                state.dismissedSecurityChecks.firmwareRevision = [];
+            if (!state.dismissedSecurityChecks.firmwareAuthenticity) {
+                state.dismissedSecurityChecks.firmwareAuthenticity = [];
             }
-            state.dismissedSecurityChecks.firmwareRevision = [
-                payload,
-                ...state.dismissedSecurityChecks.firmwareRevision,
-            ];
+            state.dismissedSecurityChecks.firmwareAuthenticity.unshift(payload);
```

Comment:
as it is wrapped in redux toolkit `createReducer`, the state here is a "draft" state. so we can safely mutate it :heavy_check_mark: